crates/ekore/src/anomalous_dimensions/unpolarized/spacelike/as4/gqg.rs

```diff
@@ -0,0 +1,166 @@
+/// Compute the singlet gluon-to-quark anomalous dimension.
+use num::complex::Complex;
+use num::traits::Pow;
+
+use crate::constants::ZETA2;
+use crate::harmonics::cache::{Cache, K};
+use crate::harmonics::log_functions::{lm11, lm12, lm13, lm14, lm14m1, lm15, lm15m1};
+
+// The routine is taken from [\[Falcioni:2023vqq\]][crate::bib:Falcioni:2023vqq].
+pub fn gamma_qg(c: &mut Cache, nf: u8, variation: u8) -> Complex<f64> {
+    let n = c.n();
+    let S1 = c.get(K::S1);
+    let S2 = c.get(K::S2);
+    let S3 = c.get(K::S3);
+    let S4 = c.get(K::S4);
+    let S5 = c.get(K::S5);
+    let nf2 = nf.pow(2) as f64;
+    let nf3 = nf.pow(3) as f64;
+
+    // Known large-x coefficients
+    let x1L5cff = 1.8518519 * nf as f64 - 4.1152263 * 0.1 * nf2;
+    let x1L4cff = 3.5687794 * 10. * nf as f64 - 3.5116598 * nf2 - 8.2304527 * 0.01 * nf3;
+    let y1L5cff = 2.8806584 * nf as f64 + 8.2304527 * 0.1 * nf2;
+    let y1L4cff = -4.0511391 * 10. * nf as f64 + 5.5418381 * nf2 + 1.6460905 * 0.1 * nf3;
+
+    // Known small-x coefficients
+    let bfkl1 = 3.9357613 * f64::pow(10., 3) * nf as f64;
+    let x0L6cff = -1.9588477 * 10. * nf as f64 + 2.7654321 * nf2;
+    let x0L5cff = 2.1573663 * 10. * nf as f64 + 1.7244444 * 10. * nf2;
+    let x0L4cff = -2.8667643 * f64::pow(10., 3) * nf as f64
+        + 3.0122403 * f64::pow(10., 2) * nf2
+        + 4.1316872 * nf3;
+
+    // The resulting part of the function
+    let P3QG01 = bfkl1 * 2. / (-1. + n).powu(3)
+        + x0L6cff * 720. / n.powu(7)
+        + x0L5cff * -120. / n.powu(6)
+        + x0L4cff * 24. / n.powu(5)
+        + x1L4cff * lm14(n, S1, S2, S3, S4)
+        + x1L5cff * lm15(n, S1, S2, S3, S4, S5)
+        + y1L4cff * lm14m1(n, S1, S2, S3, S4)
+        + y1L5cff * lm15m1(n, S1, S2, S3, S4, S5);
```

Comment:
since `lm15m1` appears both in qg and gq, maybe we should cache it ... (if you want to do a proper job, we should first check if that makes a difference at all :see_no_evil:)

Reply:
indeed will it make a difference, all the harmonics are already coming from the cache?

Reply:
yes, the harmonics are cached - maybe let's not do an optimization without checking it ... the numerical argument can be: loops and branching are expensive (as e.g. in `polygamma`), sums and product are fast (as in `lmblub`)